packages/asset/contracts/Asset.sol

```diff
@@ -385,6 +301,30 @@ contract Asset is
         return ERC1155URIStorageUpgradeable.uri(tokenId);
     }
 
+    function getTokenIdByMetadataHash(
+        string memory metadataHash
+    ) public view returns (uint256) {
+        return hashUsed[metadataHash];
+    }
+
+    function setMetadataHashUsed(
+        uint256 tokenId,
+        string memory metadataHash
+    ) public onlyRole(MINTER_ROLE) {
+        require(hashUsed[metadataHash] == 0, "metadata hash already used");
+        hashUsed[metadataHash] = tokenId;
+        _setURI(tokenId, metadataHash);
+    }
+
+    function getIncrementedCreatorNonce(
+        address creator
+    ) public onlyRole(MINTER_ROLE) returns (uint16) {
+        unchecked {
+            creatorNonces[creator]++;
+        }
+        return creatorNonces[creator];
+    }
```

Comment:
could we extract everything related to creator nonce and put it in another contract

Reply:
also it doesn't seem to be used anymore except by `mintSpecial`
but i don't understand why we need it (we have mint / mintBatch)

Reply:
We are going to need it but the `creatorNonce` will be moved to the AssetCreate contract that will come next.